bug(dashboard): WASM build fails on main (18 compile errors, not detected by CI)

### Prerequisites

- [x] I have searched existing issues to verify this bug hasn't been reported

### Description

Building the dashboard for `wasm32-unknown-unknown` fails on `main` (HEAD `6e253d5d`). Running the documented `cargo make wasm-compile-dev` from `dashboard/` (or the equivalent `cargo build --lib --target wasm32-unknown-unknown -p reinhardt-cloud-dashboard`) produces 18 compile errors and 4 warnings. The dashboard ships a WASM frontend in production, so this is a release-impacting bug that is currently invisible because no CI workflow runs the WASM build.

### Symptoms

```text
$ cd dashboard && cargo make wasm-compile-dev
...
error[E0433]: failed to resolve: could not find `db` in `reinhardt`
error[E0433]: failed to resolve: could not find `organizations` in `apps`
error[E0433]: failed to resolve: could not find `organizations` in `apps`
error[E0433]: failed to resolve: could not find `db` in `reinhardt`
error[E0433]: failed to resolve: could not find `deployments` in `apps`
error[E0432]: unresolved import `crate::apps::auth::models`
error[E0432]: unresolved import `chrono`
error[E0432]: unresolved import `tracing`
error[E0432]: unresolved import `tracing`
error[E0433]: failed to resolve: could not find `models` in `auth`
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `uuid`
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `uuid`
error[E0433]: failed to resolve: could not find `models` in `auth`
error[E0061]: this function takes 2 arguments but 3 arguments were supplied
  --> dashboard/src/apps/auth/client/pages/login.rs:17:14
error[E0061]: this function takes 3 arguments but 4 arguments were supplied
  --> dashboard/src/apps/auth/client/pages/register.rs:18:14
error[E0282]: type annotations needed
  --> dashboard/src/apps/auth/server/register.rs:166:20
error: could not compile `reinhardt-cloud-dashboard` (lib) due to 18 previous errors
```

### Root cause categories

1. **Server-only modules referenced from WASM build path.** `apps/auth/server/{login,register}.rs`, `apps/organizations/...`, and `apps/deployments/...` pull in `reinhardt::db`, `chrono`, `tracing`, and `uuid`, all of which are declared under `[target.'cfg(not(target_arch = \"wasm32\"))'.dependencies]` in `dashboard/Cargo.toml`. The modules themselves are not consistently gated with `#[cfg(not(target_arch = \"wasm32\"))]` (or the equivalent `#[cfg(native)]` alias defined in `dashboard/build.rs`), so they get compiled when targeting wasm and fail to resolve their imports.

2. **`form!` macro / server_fn signature drift.** `apps/auth/client/pages/login.rs:17` and `register.rs:18` invoke `form! { server_fn: login, ... }` / `form! { server_fn: register, ... }`, but the corresponding server functions take a different number of arguments than the macro is generating call sites for. The 4 deprecation warnings about implicit CSRF token injection (see reinhardt-web#3971) point at the same area: the `csrf_token: String` argument is now expected to be declared explicitly and threaded via `strip_arguments`, and the current call sites still rely on the deprecated implicit injection — but a downstream change appears to have produced an off-by-one in the generated argument list.

3. **Inference site missing on register error path.** `apps/auth/server/register.rs:166` — `let err_lower = e.to_string().to_lowercase();` — fails with E0282 because `e` is being treated as an unknown error type when compiled in this code path. (Even though this file should be gated for wasm, it is currently being compiled and exposed by the same gating gap as item 1.)

### Why this is invisible to CI

`grep -lriE 'wasm|wasm32' .github/` returns no matches. The repository runs unit tests, integration tests, fmt, clippy, codeql, audit, and several other workflows on every PR, but no job builds the dashboard for `wasm32-unknown-unknown`. The breakage therefore lands silently and is only noticed when a developer tries to ship the WASM bundle locally.

### Reproduction

```bash
git checkout main           # 6e253d5d at the time of filing
cd dashboard
cargo make wasm-compile-dev
# 18 errors + 4 warnings
```

### Suggested fixes

1. **Gate the server-only modules.** Audit `dashboard/src/apps/{auth,organizations,deployments}` and apply `#[cfg(native)]` (or `#[cfg(not(target_arch = \"wasm32\"))]`) to every module / item that depends on `reinhardt::db`, `chrono`, `tracing`, `uuid`, or `crate::apps::*::models`. The `cfg_aliases` block in `dashboard/build.rs` already defines `wasm` / `native`, so the gating can be uniform.
2. **Fix the `form!` macro call sites.** Update `apps/auth/client/pages/{login,register}.rs` to explicitly declare `csrf_token: String` on the corresponding server_fn signatures and pass it via `strip_arguments: { csrf_token: ::reinhardt::reinhardt_pages::csrf::get_csrf_token().unwrap_or_default() }`, mirroring the migration described in reinhardt-web#3971. This will both eliminate the E0061 errors and clear the deprecation warnings.
3. **Add a WASM build job to CI.** A minimal `wasm-build.yml` workflow that runs `cargo make wasm-compile-dev` from `dashboard/` on every PR would prevent this kind of breakage from regressing again. Even just a `cargo check --target wasm32-unknown-unknown -p reinhardt-cloud-dashboard` step would catch all 18 errors above.

### Scope

This is a Reinhardt Cloud bug, not an upstream `reinhardt-web` issue — the missing `cfg` gating and stale `form!` migration both live in this repository. Item 2 may benefit from cross-referencing reinhardt-web#3971 for the migration recipe but does not need to be reported upstream.

### Discovered while

Verifying #442 (PR #445), which adopts `reinhardt::pages::ClientLauncher` in the WASM entry. The `cargo make wasm-compile-dev` baseline failure was already present on `main` before that PR, and the diff there is entirely under `#[cfg(wasm)]` and so does not affect this issue's reproduction.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/dashboard/src/apps.rs
+++ b/dashboard/src/apps.rs
@@ -7,7 +7,10 @@
 pub mod auth;
 #[cfg(native)]
 pub mod clusters;
-#[cfg(native)]
+// `deployments` is server-only at the module level, but it owns a `client`
+// submodule containing WASM UI components consumed by `crate::client::ws`.
+// The cfg gating that excludes server-only sources lives inside
+// `apps/deployments.rs`, so the parent module declaration is unconditional.
 pub mod deployments;
 #[cfg(native)]
 pub mod health;

--- a/dashboard/src/apps/auth/client/pages/login.rs
+++ b/dashboard/src/apps/auth/client/pages/login.rs
@@ -48,6 +48,12 @@ pub fn login_page() -> Page {
 			},
 			submit: SubmitButton { label: "Sign in", class: "btn-primary w-full py-2.5 text-base" },
 		},
+		// Explicit CSRF wiring (reinhardt-web#3971) — reads the token from
+		// the cookie/meta/input chain at submit time and routes it to the
+		// server_fn's `csrf_token` parameter.
+		strip_arguments: {
+			csrf_token: ::reinhardt::reinhardt_pages::csrf::get_csrf_token().unwrap_or_default(),
+		},
 	};
 	let form_view = login_form.into_page();
 

--- a/dashboard/src/apps/auth/client/pages/register.rs
+++ b/dashboard/src/apps/auth/client/pages/register.rs
@@ -54,6 +54,12 @@ pub fn register_page() -> Page {
 			},
 			submit: SubmitButton { label: "Create account", class: "btn-primary w-full py-2.5 text-base" },
 		},
+		// Explicit CSRF wiring (reinhardt-web#3971) — reads the token from
+		// the cookie/meta/input chain at submit time and routes it to the
+		// server_fn's `csrf_token` parameter.
+		strip_arguments: {
+			csrf_token: ::reinhardt::reinhardt_pages::csrf::get_csrf_token().unwrap_or_default(),
+		},
 	};
 	let form_view = register_form.into_page();
 

--- a/dashboard/src/apps/auth/server/login.rs
+++ b/dashboard/src/apps/auth/server/login.rs
@@ -14,6 +14,7 @@ use crate::shared::AuthResponse;
 pub async fn login(
 	username: String,
 	password: String,
+	csrf_token: String,
 	#[inject] http_request: reinhardt::pages::server_fn::ServerFnRequest,
 ) -> Result<AuthResponse, ServerFnError> {
 	#[cfg(native)]
@@ -22,6 +23,10 @@ pub async fn login(
 
 		use crate::apps::auth::services;
 		use crate::shared::UserInfo;
+
+		// CSRF validation runs in middleware before this handler, so the
+		// token is consumed only to satisfy the form! contract here.
+		let _ = csrf_token;
 
 		let user = services::verify_credentials(&username, &password)
 			.await
@@ -62,7 +67,7 @@ pub async fn login(
 		// The #[server_fn] macro replaces this body with an HTTP POST stub on
 		// wasm; this branch exists only so the function compiles as a single
 		// declaration on both targets.
-		let _ = (username, password, http_request);
+		let _ = (username, password, csrf_token, http_request);
 		unreachable!("server_fn body is replaced on wasm")
 	}
 }

--- a/dashboard/src/apps/auth/server/login.rs
+++ b/dashboard/src/apps/auth/server/login.rs
@@ -2,7 +2,7 @@
 
 use reinhardt::pages::server_fn::{ServerFnError, server_fn};
 
-use crate::shared::{AuthResponse, UserInfo};
+use crate::shared::AuthResponse;
 
 /// Authenticate user with credentials and set session cookie.
 ///
@@ -16,41 +16,53 @@ pub async fn login(
 	password: String,
 	#[inject] http_request: reinhardt::pages::server_fn::ServerFnRequest,
 ) -> Result<AuthResponse, ServerFnError> {
-	use tracing::error;
+	#[cfg(native)]
+	{
+		use tracing::error;
 
-	use crate::apps::auth::services;
+		use crate::apps::auth::services;
+		use crate::shared::UserInfo;
 
-	let user = services::verify_credentials(&username, &password)
-		.await
-		.map_err(|err| {
-			// Log internal errors for operational visibility while keeping
-			// the client-facing message generic to prevent information leakage.
-			let msg = err.to_string();
-			if msg != "Invalid credentials" {
-				error!("verify_credentials internal error: {msg}");
-				return ServerFnError::application("Internal server error");
-			}
-			ServerFnError::application("Invalid credentials")
+		let user = services::verify_credentials(&username, &password)
+			.await
+			.map_err(|err| {
+				// Log internal errors for operational visibility while keeping
+				// the client-facing message generic to prevent information leakage.
+				let msg = err.to_string();
+				if msg != "Invalid credentials" {
+					error!("verify_credentials internal error: {msg}");
+					return ServerFnError::application("Internal server error");
+				}
+				ServerFnError::application("Invalid credentials")
+			})?;
+
+		let session_id = services::create_session(&user).await.map_err(|err| {
+			error!("Failed to create session: {err}");
+			ServerFnError::application("Internal server error")
 		})?;
 
-	let session_id = services::create_session(&user).await.map_err(|err| {
-		error!("Failed to create session: {err}");
-		ServerFnError::application("Internal server error")
-	})?;
+		// Set session cookie via the SharedResponseCookies jar.
+		// The server_fn router reads SharedResponseCookies after the handler
+		// and applies them as Set-Cookie response headers.
+		let is_debug = crate::config::settings::get_settings().core.debug;
+		let secure_flag = if is_debug { "" } else { "; Secure" };
+		let cookie = format!(
+			"sessionid={session_id}; HttpOnly; SameSite=Lax; Path=/{secure_flag}; Max-Age=86400"
+		);
+		http_request.add_response_cookie(cookie);
 
-	// Set session cookie via the SharedResponseCookies jar.
-	// The server_fn router reads SharedResponseCookies after the handler
-	// and applies them as Set-Cookie response headers.
-	let is_debug = crate::config::settings::get_settings().core.debug;
-	let secure_flag = if is_debug { "" } else { "; Secure" };
-	let cookie = format!(
-		"sessionid={session_id}; HttpOnly; SameSite=Lax; Path=/{secure_flag}; Max-Age=86400"
-	);
-	http_request.add_response_cookie(cookie);
-
-	let user_info = UserInfo::from(&user);
-	Ok(AuthResponse {
-		success: true,
-		user: Some(user_info),
-	})
+		let user_info = UserInfo::from(&user);
+		Ok(AuthResponse {
+			success: true,
+			user: Some(user_info),
+		})
+	}
+	#[cfg(wasm)]
+	{
+		// The #[server_fn] macro replaces this body with an HTTP POST stub on
+		// wasm; this branch exists only so the function compiles as a single
+		// declaration on both targets.
+		let _ = (username, password, http_request);
+		unreachable!("server_fn body is replaced on wasm")
+	}
 }

--- a/dashboard/src/apps/auth/server/logout.rs
+++ b/dashboard/src/apps/auth/server/logout.rs
@@ -11,38 +11,49 @@ use reinhardt::pages::server_fn::{ServerFnError, server_fn};
 pub async fn logout(
 	#[inject] http_request: reinhardt::pages::server_fn::ServerFnRequest,
 ) -> Result<bool, ServerFnError> {
-	use tracing::warn;
-
-	use crate::apps::auth::services;
-
-	// Extract session ID from the Cookie header
-	let session_id = http_request
-		.inner()
-		.headers
-		.get("Cookie")
-		.and_then(|v| v.to_str().ok())
-		.and_then(|cookies| {
-			cookies.split(';').find_map(|pair| {
-				let pair = pair.trim();
-				let (name, value) = pair.split_once('=')?;
-				if name.trim() == "sessionid" {
-					Some(value.trim().to_string())
-				} else {
-					None
-				}
-			})
-		});
-
-	// Destroy the session in Redis if a session cookie was present
-	if let Some(ref sid) = session_id
-		&& let Err(e) = services::destroy_session(sid).await
+	#[cfg(native)]
 	{
-		warn!("Failed to destroy session during logout: {e}");
+		use tracing::warn;
+
+		use crate::apps::auth::services;
+
+		// Extract session ID from the Cookie header
+		let session_id = http_request
+			.inner()
+			.headers
+			.get("Cookie")
+			.and_then(|v| v.to_str().ok())
+			.and_then(|cookies| {
+				cookies.split(';').find_map(|pair| {
+					let pair = pair.trim();
+					let (name, value) = pair.split_once('=')?;
+					if name.trim() == "sessionid" {
+						Some(value.trim().to_string())
+					} else {
+						None
+					}
+				})
+			});
+
+		// Destroy the session in Redis if a session cookie was present
+		if let Some(ref sid) = session_id
+			&& let Err(e) = services::destroy_session(sid).await
+		{
+			warn!("Failed to destroy session during logout: {e}");
+		}
+
+		// Clear the cookie regardless of whether destruction succeeded
+		let cookie = "sessionid=; HttpOnly; SameSite=Lax; Path=/; Max-Age=0".to_string();
+		http_request.add_response_cookie(cookie);
+
+		Ok(true)
 	}
-
-	// Clear the cookie regardless of whether destruction succeeded
-	let cookie = "sessionid=; HttpOnly; SameSite=Lax; Path=/; Max-Age=0".to_string();
-	http_request.add_response_cookie(cookie);
-
-	Ok(true)
+	#[cfg(wasm)]
+	{
+		// The #[server_fn] macro replaces this body with an HTTP POST stub on
+		// wasm; this branch exists only so the function compiles as a single
+		// declaration on both targets.
+		let _ = http_request;
+		unreachable!("server_fn body is replaced on wasm")
+	}
 }

--- a/dashboard/src/apps/auth/server/register.rs
+++ b/dashboard/src/apps/auth/server/register.rs
@@ -5,7 +5,7 @@
 
 use reinhardt::pages::server_fn::{ServerFnError, server_fn};
 
-use crate::shared::{AuthResponse, UserInfo};
+use crate::shared::AuthResponse;
 
 /// Create a new user account with email verification.
 ///
@@ -20,105 +20,117 @@ pub async fn register(
 	password: String,
 	#[inject] _http_request: reinhardt::pages::server_fn::ServerFnRequest,
 ) -> Result<AuthResponse, ServerFnError> {
-	use reinhardt::BaseUser;
-	use reinhardt::db::orm::Model;
-	use tracing::{error, info};
-
-	use crate::apps::auth::models::User;
-	use crate::apps::auth::services::email::{get_email_backend, send_verification_email};
-	use crate::apps::auth::services::registration::provision_personal_organization;
-	use crate::apps::auth::services::token::{TokenPurpose, generate_token};
-
-	let settings = crate::config::settings::get_settings();
-	let secret_key = settings.core.secret_key.clone();
-	let from_email = settings.email.from_email.clone();
-
-	// Create user as inactive — requires email verification to activate
-	let mut user = User::new(
-		username.trim().to_string(),
-		email.trim().to_lowercase(),
-		String::new(),
-		String::new(),
-		None,
-		false,
-		false,
-		false,
-	);
-	user.set_password(&password).map_err(|e| {
-		error!("Password hashing failed during registration: {e}");
-		ServerFnError::application("Internal server error")
-	})?;
-
-	// Attempt to create -- database unique constraint prevents duplicates
-	let created = match User::objects().create(&user).await {
-		Ok(user) => user,
-		Err(e) => {
-			let err_lower = e.to_string().to_lowercase();
-			if err_lower.contains("unique") || err_lower.contains("duplicate") {
-				let message = if err_lower.contains("auth_user_email_uniq") {
-					"Email already exists"
-				} else {
-					"Username already exists"
-				};
-				return Err(ServerFnError::application(message));
-			}
-			error!("Failed to create user in database: {e}");
-			return Err(ServerFnError::application("Internal server error"));
-		}
-	};
-
-	// Provision a Personal Organization for the new user (refs #415, #435).
-	// Shared with the REST register flow; slug derivation, retry, and
-	// rollback are handled by the shared service.
-	provision_personal_organization(&created)
-		.await
-		.map_err(|e| ServerFnError::application(e.to_string()))?;
-
-	// Generate verification token and send email
-	let token = generate_token(
-		TokenPurpose::EmailVerification,
-		&created.id,
-		"",
-		&secret_key,
-	);
-
-	let port = std::env::var("PORT").unwrap_or_else(|_| "8000".to_string());
-	let base_url = std::env::var("REINHARDT_CLOUD_BASE_URL")
-		.unwrap_or_else(|_| format!("http://localhost:{port}"));
-	let verification_url = format!("{base_url}/api/auth/verify-email/{token}/");
-
-	let backend = get_email_backend().map_err(|e| {
-		error!("Failed to create email backend: {e}");
-		ServerFnError::application("Internal server error")
-	})?;
-
-	if let Err(e) = send_verification_email(
-		&created.email,
-		&created.username,
-		&verification_url,
-		backend.as_ref(),
-		&from_email,
-	)
-	.await
+	#[cfg(native)]
 	{
-		error!(
-			"Failed to send verification email to {}: {e}",
-			created.email
-		);
-		// Roll back user creation to avoid stranding an inactive account
-		if let Err(del_err) = User::objects().delete(created.id).await {
-			error!("Failed to roll back user after email failure: {del_err}");
-		}
-		return Err(ServerFnError::application(
-			"Registration failed — please try again later",
-		));
-	}
-	info!("Verification email sent to {}", created.email);
+		use reinhardt::BaseUser;
+		use reinhardt::db::orm::Model;
+		use tracing::{error, info};
 
-	// No session cookie — user must verify email first
-	let user_info = UserInfo::from(&created);
-	Ok(AuthResponse {
-		success: true,
-		user: Some(user_info),
-	})
+		use crate::apps::auth::models::User;
+		use crate::apps::auth::services::email::{get_email_backend, send_verification_email};
+		use crate::apps::auth::services::registration::provision_personal_organization;
+		use crate::apps::auth::services::token::{TokenPurpose, generate_token};
+		use crate::shared::UserInfo;
+
+		let settings = crate::config::settings::get_settings();
+		let secret_key = settings.core.secret_key.clone();
+		let from_email = settings.email.from_email.clone();
+
+		// Create user as inactive — requires email verification to activate
+		let mut user = User::new(
+			username.trim().to_string(),
+			email.trim().to_lowercase(),
+			String::new(),
+			String::new(),
+			None,
+			false,
+			false,
+			false,
+		);
+		user.set_password(&password).map_err(|e| {
+			error!("Password hashing failed during registration: {e}");
+			ServerFnError::application("Internal server error")
+		})?;
+
+		// Attempt to create -- database unique constraint prevents duplicates
+		let created = match User::objects().create(&user).await {
+			Ok(user) => user,
+			Err(e) => {
+				let err_lower = e.to_string().to_lowercase();
+				if err_lower.contains("unique") || err_lower.contains("duplicate") {
+					let message = if err_lower.contains("auth_user_email_uniq") {
+						"Email already exists"
+					} else {
+						"Username already exists"
+					};
+					return Err(ServerFnError::application(message));
+				}
+				error!("Failed to create user in database: {e}");
+				return Err(ServerFnError::application("Internal server error"));
+			}
+		};
+
+		// Provision a Personal Organization for the new user (refs #415, #435).
+		// Shared with the REST register flow; slug derivation, retry, and
+		// rollback are handled by the shared service.
+		provision_personal_organization(&created)
+			.await
+			.map_err(|e| ServerFnError::application(e.to_string()))?;
+
+		// Generate verification token and send email
+		let token = generate_token(
+			TokenPurpose::EmailVerification,
+			&created.id,
+			"",
+			&secret_key,
+		);
+
+		let port = std::env::var("PORT").unwrap_or_else(|_| "8000".to_string());
+		let base_url = std::env::var("REINHARDT_CLOUD_BASE_URL")
+			.unwrap_or_else(|_| format!("http://localhost:{port}"));
+		let verification_url = format!("{base_url}/api/auth/verify-email/{token}/");
+
+		let backend = get_email_backend().map_err(|e| {
+			error!("Failed to create email backend: {e}");
+			ServerFnError::application("Internal server error")
+		})?;
+
+		if let Err(e) = send_verification_email(
+			&created.email,
+			&created.username,
+			&verification_url,
+			backend.as_ref(),
+			&from_email,
+		)
+		.await
+		{
+			error!(
+				"Failed to send verification email to {}: {e}",
+				created.email
+			);
+			// Roll back user creation to avoid stranding an inactive account
+			if let Err(del_err) = User::objects().delete(created.id).await {
+				error!("Failed to roll back user after email failure: {del_err}");
+			}
+			return Err(ServerFnError::application(
+				"Registration failed — please try again later",
+			));
+		}
+		info!("Verification email sent to {}", created.email);
+
+		// No session cookie — user must verify email first
+		let user_info = UserInfo::from(&created);
+		Ok(AuthResponse {
+			success: true,
+			user: Some(user_info),
+		})
+	}
+	#[cfg(wasm)]
+	{
+		// The #[server_fn] macro replaces this body with an HTTP POST stub on
+		// wasm; this branch exists only so the function compiles as a single
+		// declaration on both targets.
+		let _ = (username, email, password, _http_request);
+		unreachable!("server_fn body is replaced on wasm")
+	}
 }

--- a/dashboard/src/apps/auth/server/register.rs
+++ b/dashboard/src/apps/auth/server/register.rs
@@ -18,6 +18,7 @@ pub async fn register(
 	username: String,
 	email: String,
 	password: String,
+	csrf_token: String,
 	#[inject] _http_request: reinhardt::pages::server_fn::ServerFnRequest,
 ) -> Result<AuthResponse, ServerFnError> {
 	#[cfg(native)]
@@ -31,6 +32,10 @@ pub async fn register(
 		use crate::apps::auth::services::registration::provision_personal_organization;
 		use crate::apps::auth::services::token::{TokenPurpose, generate_token};
 		use crate::shared::UserInfo;
+
+		// CSRF validation runs in middleware before this handler, so the
+		// token is consumed only to satisfy the form! contract here.
+		let _ = csrf_token;
 
 		let settings = crate::config::settings::get_settings();
 		let secret_key = settings.core.secret_key.clone();
@@ -130,7 +135,7 @@ pub async fn register(
 		// The #[server_fn] macro replaces this body with an HTTP POST stub on
 		// wasm; this branch exists only so the function compiles as a single
 		// declaration on both targets.
-		let _ = (username, email, password, _http_request);
+		let _ = (username, email, password, csrf_token, _http_request);
 		unreachable!("server_fn body is replaced on wasm")
 	}
 }

--- a/dashboard/src/apps/deployments.rs
+++ b/dashboard/src/apps/deployments.rs
@@ -1,16 +1,29 @@
 //! Deployments application module.
 //!
-//! Provides endpoints for application deployment management.
+//! Provides endpoints for application deployment management on the server,
+//! and a `client` submodule containing the WASM UI components for the
+//! real-time log viewer and cluster health panel.
 
+#[cfg(native)]
 use reinhardt::app_config;
 
-pub mod admin;
+// The `client` submodule is consumed from the WASM build path and is
+// therefore unconditional. Its own internals are wasm-gated as needed.
 pub mod client;
+
+#[cfg(native)]
+pub mod admin;
+#[cfg(native)]
 pub mod models;
+#[cfg(native)]
 pub mod serializers;
+#[cfg(native)]
 pub mod tests;
+#[cfg(native)]
 pub mod urls;
+#[cfg(native)]
 pub mod views;
 
+#[cfg(native)]
 #[app_config(name = "deployments", label = "deployments")]
 pub struct DeploymentsConfig;

